homeassistant/components/local_calendar/__init__.py

```diff
@@ -42,6 +46,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except OSError as err:
         raise ConfigEntryNotReady("Failed to load file {path}: {err}") from err
 
+    if entry.data.get(CONF_ICS_FILE):
+        with process_uploaded_file(hass, entry.data[CONF_ICS_FILE]) as ics_file:
```

Comment:
Right now `CONF_STORAGE_KEY` refers to a specific ics path above based on `STORAGE_PATH`.  (In other words `CONF_STORAGE_KEY` is effectively already `CONF_ICS_FILE` but limited to a specific location.)

Can the config flow place the file into that destination? 

This can also allow validation at upload time to verify the file parses when its moved into the final location.  This also can let us import from a URL as well and have the same behavior.

Reply:
That makes sense, I think I've reworked what I had to instead move the file to where `CONF_STORAGE_KEY` points. I also see how this will allow for easier implementation of importing from a URL.